### PR TITLE
feat: [PL-39354]: proxy stackdriver logs logging service

### PR DESCRIPTION
**Pull request by mdmilic**
## Describe your changes
* Create endpoint on logging service which will receive a collection of log lines
* Write all the lines received on the above endpoint to local stackdriver from the cluster where logging service is running
* This feature is optional and disabled for SMP (i.e. if GCP projectId not provided, the endpoint will not be mounted)

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53388)

### CODE REVIEW

**Comment by mdmilic** [comment]
trigger ut8

File: product/log-service/handler/stackdriver.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"cloud.google.com/go/logging"
+	"encoding/json"
+	"github.com/harness/harness-core/product/log-service/logger"
+	"github.com/harness/harness-core/product/log-service/stackdriver"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	"net/http"
+)
+
+// HandlePost returns a http.HandlerFunc that writes log lines from payload to stackdriver.
+func HandlePost(s *stackdriver.Stackdriver) http.HandlerFunc {
```
Comment (smjt-h) [inline product/log-service/handler/stackdriver.go:18]: Can we rename this func that serves the purpose something like HandleStackDriverWrite()

File: product/log-service/config/config.go [hunk 2]
```diff
@@ -68,6 +68,10 @@ type Config struct {
 		ConsumerGroup string `envconfig:"LOG_SERVICE_CONSUMER_GROUP" default:"group-one"`
 	}
 
+	Stackdriver struct {
+		ProjectID string `envconfig:"LOG_SERVICE_STACKDRIVER_PROJECT_ID"`
```
Comment (xingchi-jin) [inline product/log-service/config/config.go:72]: what is the project id ?

File: product/log-service/stackdriver/stackdriver.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	return s.client.Ping(ctx)
+}
+
+func (s *Stackdriver) Close() error {
+	return s.client.Close()
+}
+
+// Line represents a structured line in the stackdriver logs.
+type Line struct {
+	Labels   map[string]string      `json:"labels"`
```
Comment (xingchi-jin) [inline product/log-service/stackdriver/stackdriver.go:53]: will there be nested labels ?

**Review by xingchi-jin — APPROVED** [review]

File: product/log-service/config/config.go (hunk 2 above)
Comment (mdmilic) [reply product/log-service/config/config.go:74]: This is the GCP projectId (e.g. `del-play`)

File: product/log-service/stackdriver/stackdriver.go (hunk 3 above)
Comment (mdmilic) [reply product/log-service/stackdriver/stackdriver.go:53]: Currently we have only a set of 6-7 static labels and they are not nested. Not sure what would be an example of nested label?

Comment (xingchi-jin) [reply product/log-service/stackdriver/stackdriver.go:53]: you are right, i just checked, we are not using nested labels for delegate logs

**Review by xingchi-jin — APPROVED** [review]

**Comment by mdmilic** [comment]
trigger uts

**Comment by mdmilic** [comment]
trigger gobuild

**Comment by gn-harness** [comment]
FMing as only unrelated tests are failing:

//400-rest:software.wings.service.impl.trigger.tests                     FAILED in 258.4s
  /tmp/execroot/harness_monorepo/bazel-out/k8-fastbuild/testlogs/400-rest/software.wings.service.impl.trigger.tests/test.log

**Pull request merged**